plugins/hypervisors/xenserver/src/main/java/org/apache/cloudstack/hypervisor/xenserver/ExtraConfigurationUtility.java

```diff
@@ -169,12 +178,28 @@ private static void applyConfigWithKeyValue(Connection conn, VM vm, Map<String,
         }
     }
 
-    private static Map<String, String> prepareKeyValuePair(String cfg) {
-        Map<String, String> configKeyPair = new HashMap<>();
+    private static Pair<String, String> prepareKeyValuePair(String cfg) {
```

Comment:
Should we add some unit tests for this methods?

Reply:
@shwstppr added unit tests